### PR TITLE
Assignment 

### PR DESCRIPTION
**Pull request by boyi01**
### Short Description
Christoph Becker has done the Assignment 

### Issue Reference
fixes #

### Checks

- [ ] This PR follows the coding guidelines of the project
- [ ] This PR fulfills the acceptance criteria defined on the issue
- [ ] This PR has been locally tested
- [ ] Changes in this PR are unit tested

### CODE REVIEW

**Comment by mnembrini** [comment]
Thanks for the submission, we will review it shortly

**Review by norbertspiess — COMMENTED** [review]
Thanks for submitting your solution. All tests pass, well done. The solution is sound and uses modern Java APIs.

Please see all my comments as remarks only and nothing to be fixed. Feel free to answer some of my questions though. I'd be curious to hear your thoughts on them.

As a side not, I'm surprised by the inconsistent code style and the provided yet unfollowed formatting. 🙁

@mnembrini will have a look as well and we'll have a chat together about your solution.

**Comment by mnembrini** [comment]
Thanks @boyi01 for that. Overall good solution, although we would have like more consistent formatting (see the assignment instructions).
Simona will contact you about the next steps

File: src/main/java/ch/aaap/assignment/Application.java [hunk 1]
```diff
@@ -4,97 +4,137 @@
...
+        return getDisctrictCodesinCaton(cantonCode).size();
+    }
+
+    /**
+     * @param districtCode of a district (e.g. 101)
+     * @return amount of districts in given canton
+     */
+    public long getAmountOfPoliticalCommunitiesInDistrict(String districtCode) {
+        Long AmountofPoliticalCommunities = Long.valueOf(0);
+        AmountofPoliticalCommunities += model.getPoliticalCommunities().stream().filter(element -> districtCode.equals(element.getDistrictCode())).count();
```
[the remark this answers is not in the document]
Comment (boyi01) [reply src/main/java/ch/aaap/assignment/Application.java:83]: After a quick review, this isn't useful at all...

File: src/main/java/ch/aaap/assignment/Application.java [hunk 2]
```diff
@@ -4,97 +4,137 @@
...
+     * @param zipCode 4 digit zip code
+     * @return district that belongs to specified zip code
+     */
+    public Set<String> getDistrictsForZipCode(String zipCode) {
+        Set<String> politicalCommunityNumbers;
+        politicalCommunityNumbers = model.getPostalCommunities().stream().filter(element -> element.getZipCode().equals(zipCode)).map(element -> element.getPoliticalCommunityNumber()).flatMap(Collection::stream).collect(Collectors.toSet());
+        Set<String> districts;
+        districts = model.getPoliticalCommunities().stream().filter(element -> politicalCommunityNumbers.contains(element.getNumber())).map(element -> element.getDistrictCode()).collect(Collectors.toSet());
+        Set<String> district_names;
+        district_names = model.getDistricts().stream().filter(element -> districts.contains(element.getNumber())).map(element -> element.getName()).collect(Collectors.toSet());
```
[the remark this answers is not in the document]
Comment (boyi01) [reply src/main/java/ch/aaap/assignment/Application.java:100]: This is one of my problems, in my daily buissnes i code also BASH and SAS and there we Use snake_case so sometime i just mess this up..

File: src/main/java/ch/aaap/assignment/raw/District.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    private String Number;
+    private String Name;
+    private String CantonCode;
+
+    public District(String number, String name, String cantonCode) {
+        Number = number;
+        Name = name;
+        CantonCode = cantonCode;
+    }
```
[the remark this answers is not in the document]
Comment (boyi01) [reply src/main/java/ch/aaap/assignment/raw/District.java:15]: I'm really used to databased styled datatypes so i guess that's why is was my first idea.

**Comment by boyi01** [comment]
Thank you for your feedback, we can have a little chat soon.
The formatting was completely done with google-java-format.... in this point, I'm kind of lazy... sad that this was not working as expected.
I really appreciate that: "modern Java APIs" cause this was the first time for me using Java above version 7.

**Pull request closed without merging**